services/file_service.py

```
@@ -27,7 +32,8 @@ def save_upload_file(upload_file: IO, filename: str) -> str:
     # Ensure final path is within UPLOAD_DIR
     upload_dir_abs = os.path.abspath(UPLOAD_DIR)
     if not final_path.startswith(upload_dir_abs + os.sep):
-        raise ValueError("Unsafe file path.")
+        logger.error("Unsafe file path detected: %s", final_path)
+        raise DocumentSaveError("Unsafe file path.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)